### ISSUE TITLE
Toggle Navigation hides links from screen reader (JAWS) link list.

### ISSUE THREAD

**Issue by kaili-kameoka** [twbs/bootstrap#36586]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I've received reports from screen reader users that the menu links that can be found visually in the desktop version of the Navbar do not appear in the "Links List" in JAWS. The links only appear when the user toggles the menu toggle button. The toggle button label is also misleading as it does not indicate that it belongs to the menu or wither the menu is open or not.

### Reduced test cases

https://guidedogsofhawaii.org/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.1.3

**Comment by patrickhlauke** [comment]
The fact that links inside the closed menu can't be found is expected, as they're fully removed from the document (with `display:none`) when closed. This is the standard way such menus are implemented, as if they weren't, they'd still be in the general focus/reading order of the page even when not visible.

The off-canvas navigation menu only appears for screen sizes below `992px`. Now, it seems that for the navigation, something's not quite right in how the off-canvas was implemented: while the off-canvas for the "Toggle docs navigation" correctly traps focus, the off-canvas for the general navigation doesn't...users can still tab out of this and into the underlying page. @mdo worth checking how this was added/implemented in this case. Also yes, while the "Toggle navigation" button does have an `aria-expanded`, it's currently always set to `false`, regardless. I suspect this was a left-over from the previous design that used dropdowns, as the off-canvas doesn't set `aria-expanded` since the idea is that it's a modal, and the underlying page should not actually be accessible (because of the focus trapping and the `aria-modal="true"`).

Also noticed that focus is not moved to either of the off-canvases (the docs nav, and the general nav), but remains on the toggle buttons. Not looked at the markup in depth, but suspect a `tabindex="-1"` is missing on them (compare to the working off-canvas examples https://getbootstrap.com/docs/5.2/components/offcanvas/).

In short, the fact the links aren't showing up in link list is expected. But the rest is down to incorrect implementation of the off-canvas used.

**Comment by patrickhlauke** [comment]
if the problems you reported were about https://guidedogsofhawaii.org/ - again, the fact that links in closed menus/dropdowns aren't in the JAWS link list is expected, as they're completely hidden both visually and form assistive technologies.

however, not seeing an issue regarding the toggle button not reporting its state: it does announce whether the expand/collapse panel it controls is expanded or collapsed - "Toggle navigation button, collapsed" / "Toggle navigation button, expanded" (Tested in Chrome/JAWS 2022)

**Comment by mdo** [comment]
Closing per above comment (I think I understand that we're not seeing an issue).
